impeller/entity/contents/filters/morphology_filter_contents.cc

```diff
@@ -160,8 +160,25 @@ std::optional<Rect> DirectionalMorphologyFilterContents::GetFilterCoverage(
   auto transformed_vector =
       transform.TransformDirection(direction_ * radius_.radius).Abs();
 
-  auto extent = coverage->size + transformed_vector * 2;
-  return Rect(coverage->origin - transformed_vector, Size(extent.x, extent.y));
+  auto origin = coverage->origin;
+  auto width = coverage->size.width;
+  auto height = coverage->size.height;
+  switch (morph_type_) {
+    case FilterContents::MorphType::kDilate:
+      origin -= transformed_vector;
+      width += transformed_vector.x * 2;
+      height += transformed_vector.y * 2;
```

Comment:
Nit: The temp width/height could be a Vector2, which would allow this arithmetic to be done in fewer lines at no cost.
```
size += transformed_vector * 2;
```